app/lib/service/openid/github_jwks.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:pub_dev/shared/redis_cache.dart';
+import 'package:pub_dev/tool/utils/http.dart';
+
+import 'jwks.dart';
+
+/// Fetches the OpenID configuration and then the JSON Web Key list from GitHub.
+Future<JsonWebKeyList> fetchGithubOpenIdWebKeyList() async {
+  final list = await cache.githubOpenIdWebKeyList().get(() async {
+    final client = httpRetryClient();
+    try {
+      final configUri = Uri.parse(
+          'https://token.actions.githubusercontent.com/.well-known/openid-configuration');
+      final rs1 = await client.get(configUri);
+      if (rs1.statusCode != 200) {
+        throw Exception(
+            'Unexpected status code ${rs1.statusCode} while fetching $configUri');
+      }
+      final map1 = json.decode(rs1.body) as Map<String, dynamic>;
+      final jwksUri = Uri.parse(map1['jwks_uri'] as String);
```

Comment:
may I propose we make an `OpenIDProviderConfiguration` or `OpenIdProviderMetadata` object too?

If we look at what github supplies:
```js
{
  // We should require that "issuer" matches the "iss" claim in the signed JWTs
  // we could even require that the issuer in this document matches something hardcoded in our code...
  "issuer": "https://token.actions.githubusercontent.com",
  // let's sanity check that jwks_uri is an HTTPS URL :D 
  "jwks_uri": "https://token.actions.githubusercontent.com/.well-known/jwks",
  "subject_types_supported": [
    "public",
    "pairwise"
  ],
  "response_types_supported": [
    "id_token" // we should require that this includes "id_token"
  ],
  "claims_supported": [
    // We should reject any JWT with claims not listed here:
    "sub",
    "aud",
    "exp",
    "iat",
    "iss",
    "jti",
    "nbf",
    "ref",
    "repository",
    "repository_id",
    "repository_owner",
    "repository_owner_id",
    "run_id",
    "run_number",
    "run_attempt",
    "actor",
    "actor_id",
    "workflow",
    "head_ref",
    "base_ref",
    "event_name",
    "ref_type",
    "environment",
    "job_workflow_ref"
  ],
  "id_token_signing_alg_values_supported": [
    // we should:
    //  * reject any JWK for other algorithms,
    //  * reject any JWT for other algorithms,
    "RS256"
  ],
  "scopes_supported": [
    "openid" // I guess we could require this includes "openid", spec says so: https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata
  ]
}
```

We can probably just make a generic structure to decode:
https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata

And then a method / extension method to validate this against the specification.

And then a function to check that a given `OpenIDProviderConfiguration` object satisfies our hardcoded requirements for github actions.

Maybe we can create a `JwtValidator` object that takes a `OpenIDProviderConfiguration` and `jwkKeys` in the constructor and offers a mechanism to validate the JWTs. Just thinking... Just saying, we probably need to decode and cache the entire openid-configuration object.